internal/cli/opsmanager/backup/snapshots/schedule/update.go

```diff
@@ -0,0 +1,125 @@
+	clusterID                      string
+	referenceTimeZoneOffset        string
+	dailySnapshotRetentionDays     int
+	clusterCheckpointIntervalMin   int
+	monthlySnapshotRetentionMonths int
+	snapshotIntervalHours          int
+	snapshotRetentionDays          int
+	weeklySnapshotRetentionWeeks   int
+	pointInTimeWindowHours         int
+	referenceHourOfDay             int
+	referenceMinuteOfHour          int
+}
+
+func (opts *UpdateOpts) init() error {
+	var err error
+	opts.store, err = store.New(config.Default())
+	return err
+}
+
+func (opts *UpdateOpts) Run() error {
+	r, err := opts.store.UpdateSnapshotSchedule(opts.ConfigProjectID(), opts.clusterID, opts.newSnapshotSchedule())
+	if err != nil {
+		return err
+	}
+	return opts.Print(r)
+}
+
+func (opts *UpdateOpts) newSnapshotSchedule() *opsmngr.SnapshotSchedule {
+	snapshotSchedule := &opsmngr.SnapshotSchedule{
+		ClusterID:                      opts.clusterID,
+		GroupID:                        opts.ConfigProjectID(),
+		ReferenceTimeZoneOffset:        opts.referenceTimeZoneOffset,
+		DailySnapshotRetentionDays:     opts.dailySnapshotRetentionDays,
+		ClusterCheckpointIntervalMin:   opts.clusterCheckpointIntervalMin,
+		MonthlySnapshotRetentionMonths: opts.monthlySnapshotRetentionMonths,
+		SnapshotIntervalHours:          opts.snapshotIntervalHours,
+		SnapshotRetentionDays:          opts.snapshotRetentionDays,
+		WeeklySnapshotRetentionWeeks:   opts.weeklySnapshotRetentionWeeks,
+	}
+
+	if opts.pointInTimeWindowHours >= 0 {
+		snapshotSchedule.PointInTimeWindowHours = &opts.pointInTimeWindowHours
+	}
+
+	if opts.referenceHourOfDay >= 0 {
+		snapshotSchedule.ReferenceHourOfDay = &opts.referenceHourOfDay
+	}
+
+	if opts.referenceMinuteOfHour >= 0 {
+		snapshotSchedule.ReferenceMinuteOfHour = &opts.referenceMinuteOfHour
+	}
+
+	return snapshotSchedule
+}
+
+// mongocli ops-manager backup snapshot(s) schedule update [--clusterID clusterID ][--projectId projectId] [--referenceTimeZoneOffset referenceTimeZoneOffset]
+// [--dailySnapshotRetentionDays dailySnapshotRetentionDays] [--clusterCheckpointIntervalMin clusterCheckpointIntervalMin]
+// [--snapshotIntervalHours snapshotIntervalHours] [--snapshotRetentionDays snapshotRetentionDays] [--weeklySnapshotRetentionWeeks weeklySnapshotRetentionWeeks]
+// [--pointInTimeWindowHours pointInTimeWindowHours] [--referenceHourOfDay referenceHourOfDay] [--referenceMinuteOfHour referenceMinuteOfHour]
+// [--monthlySnapshotRetentionMonths monthlySnapshotRetentionMonths]
+
+func UpdateBuilder() *cobra.Command {
+	opts := &UpdateOpts{}
+	opts.Template = updateTemplate
+	cmd := &cobra.Command{
+		Use:   "update",
+		Short: update,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.OutWriter = cmd.OutOrStdout()
+			return opts.init()
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.clusterID, flag.ClusterID, "", usage.ClusterID)
```

Comment:
this should be required, aren't any other flags required here?

Reply:
no, the only required parameters are projectID and clusterID https://docs.opsmanager.mongodb.com/current/reference/api/backup/update-one-snapshot-schedule-by-cluster-id/